client/src/components/app/App.js

```diff
@@ -1,10 +1,95 @@
 import React from "react";
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
+import 'antd/dist/antd.css';
 
+import {
+  Home,
+  Login,
+  CreateProfile,
+  Posts,
+  Post,
+  PostForm,
+  PublicService,
+  PageNotFound,
+  Event,
+} from 'components/pages';
+import { Header, Footer } from 'components/utils'
 import "./App.css";
 
 function App() {
-  return <>
-  </>;
+  return (
+    <Router>
+      <Header />
+      <Switch>
+        <Route
+          exact
+          path='/'
+          component={Home}
+        />
+        <Route
+          path='/signup'
+          component={CreateProfile}
+        />
+        <Route
+          path='/login'
+          component={Login}
+        />
+        {/* <Route
+            exact
+            path='/profile/:id'
+            component={Profile}
+          /> */}
+        <Route
+          path='/posts/live'
+          render={(props) => <Post {...props} postType='live' />}
+        />
+        <Route
+          path='/posts/draft'
+          render={(props) => <Post {...props} postType='draft' />}
+        />
+        <Route
+          path='/posts/new'
+          component={PostForm}
+        />
+        <Route
+          exact
+          path='/post/public-service/:id/edit'
+          render={(props) => <PostForm postFormType='public service' {...props} />}
+        />
+        <Route
+          exact
+          path='/post/event/:id/edit'
+          render={(props) => <PostForm postFormType='event' {...props} />}
+        />
+        <Route
+          path='/post/event/:category/:id'
+          render={(props) => <Event {...props} postStatus='published' />}
+        />
+        <Route
+          path='/post/public-service/:category/:id'
+          render={(props) => <PublicService {...props} postStatus='published' />}
+        />
+        <Route
+          path='/post/event/:category/:id/preview'
```

Comment:
@IsraaSulaiman What is the difference between the page shown for "/post/event/:category/:id/preview" and the page shown for "/post/event/:category/:id" (the route on line 64)?  They look like they both show the same thing, in their render:
`render={(props) => <Event {...props} postStatus='published' />}`
So maybe you could remove the preview route if it's redundant.  I think the same is true for "/post/public-service/:category/:id" and "/post/public-service/:category/:id/preview"

One other thing to keep in mind is that the router will choose the first route inside it that matches, so if you have two routes like this: 

/post/public-service/:category/:id
/post/public-service/:category/:id/preview

.. the router will take urls that end with /preview and decide that they match the first route (the one that ends in /:id) so the second preview route's component will never match and will never be shown.
You need to swap their order and put the more specific one first: 

/post/public-service/:category/:id/preview
/post/public-service/:category/:id

or alternatively, use the [exact](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/Route.md#exact-bool) prop on the first route.